packages/hdwallet-walletconnect/src/walletconnect.ts

```diff
@@ -0,0 +1,317 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import WalletConnectProvider from "@walletconnect/web3-provider";
+import isObject from "lodash/isObject";
+
+export function isWalletConnect(wallet: core.HDWallet): wallet is WalletConnectHDWallet {
+  return isObject(wallet) && (wallet as any)._isWalletConnect;
+}
+
+/**
+ * WalletConnect Wallet Info
+ * 
+ * Supported JSON-RPC API Methods:
+ * - personal_sign
+ * - eth_sign
+ * - eth_signTypedData
+ * - eth_sendTransaction
+ * - eth_signTransaction
+ * 🚧 eth_sendRawTransaction
+ * @see https://docs.walletconnect.com/
+ */
+export class WalletConnectWalletInfo implements core.HDWalletInfo, core.ETHWalletInfo {
+  readonly _supportsETHInfo = true;
+  readonly _supportsBTCInfo = false;
+  public getVendor(): string {
+    return "WalletConnect";
+  }
+
+  public hasOnDevicePinEntry(): boolean {
+    return false;
+  }
+
+  public hasOnDevicePassphrase(): boolean {
+    return false;
+  }
+
+  public hasOnDeviceDisplay(): boolean {
+    return false;
+  }
+
+  public hasOnDeviceRecovery(): boolean {
+    return false;
+  }
+
+  public hasNativeShapeShift(): boolean {
+    return false;
+  }
+
+  public supportsOfflineSigning(): boolean {
+    return false;
+  }
+
+  public supportsBroadcast(): boolean {
+    return true;
+  }
+
+  public describePath(msg: core.DescribePath): core.PathDescription {
+    switch (msg.coin) {
+      case "Ethereum":
+        return core.describeETHPath(msg.path);
+      default:
+        throw new Error("Unsupported path");
+    }
+  }
+
+  public ethNextAccountPath(_msg: core.ETHAccountPath): core.ETHAccountPath | undefined {
+    return undefined;
+  }
+
+  public async ethSupportsNetwork(chainId = 1): Promise<boolean> {
+    return chainId === 1;
+  }
+
+  public async ethSupportsSecureTransfer(): Promise<boolean> {
+    return false;
+  }
+
+  public ethSupportsNativeShapeShift(): boolean {
+    return false;
+  }
+
+  public async ethSupportsEIP1559(): Promise<boolean> {
+    return false;
+  }
+
+  public ethGetAccountPaths(msg: core.ETHGetAccountPath): Array<core.ETHAccountPath> {
+    const slip44 = core.slip44ByCoin(msg.coin);
+    if (slip44 === undefined) return [];
+    return [
+      {
+        addressNList: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx, 0, 0],
+        hardenedPath: [0x80000000 + 44, 0x80000000 + slip44, 0x80000000 + msg.accountIdx],
+        relPath: [0, 0],
+        description: "WalletConnect",
+      },
+    ];
+  }
+}
+
+export class WalletConnectHDWallet implements core.HDWallet, core.ETHWallet {
+  readonly _supportsETH = true;
+  readonly _supportsETHInfo = true;
+  readonly _supportsBTCInfo = false;
+  readonly _supportsBTC = false;
+  readonly _isWalletConnect = true;
+
+  info: WalletConnectWalletInfo & core.HDWalletInfo;
+  provider: WalletConnectProvider;
+  connected = false;
+  chainId: number = -1;
+  accounts: string[] = [];
+  ethAddress: string = "";
```

Comment:
I'm thinking maybe it might be an improvement to make these fields private. Is there any reason not to?

Reply:
If you're only talking about the four that I added, then yes, I could see it as an improvement.